### ISSUE TITLE
Add Firefox+You Newsletter Opt-in to FxA Sign-In Form

### ISSUE THREAD

**Issue by chrismore** [mozilla/fxa-features#55]
Currently, the newsletter opt-in is only on the sign up form and there are plenty of sign ins happening that may not know there is a way to hear more about how Firefox can help them online.

**Comment by seanmonstar** [comment]
Would a possible alternative be to add newsletter checkboxes in the Account Settings page (the one where you edit your avatar, devices, etc)?

**Comment by rfk** [comment]
There's already a checkbox in the settings page, but it's hidden away under "Communication preferences".  We've probably got datadog metrics on how often users view this screen in practice.

**Comment by shane-tomlinson** [comment]
On a related note, we never fixed the bug where users that sign up for the newsletter on signup but verify on a 2nd device are not actually signed up for the newsletter. We use OAuth tokens to sign up the user to the newsletter and users that verify on a 2nd device cannot generate an OAuth token.

**Comment by chrismore** [comment]
@shane-tomlinson do you have a link to that bug about the verify on a second device or know how often that happens?

**Comment by chrismore** [comment]
@shane-tomlinson found the bug: https://bugzilla.mozilla.org/show_bug.cgi?id=1281930

@davismtl given that the newsletters and the multi-device sync journeys are something a bunch of us are spending quality time on getting right, what is your thoughts on prioritizing this bug at some point? Given that so many people verify on other devices, seems like this could be a win for people who have actively said they want to have a deeper relationship with Firefox.

**Comment by shane-tomlinson** [comment]
Our GitHub copy of the referenced bugzilla bug is https://github.com/mozilla/fxa-bugzilla-mirror/issues/167

**Comment by davismtl** [comment]
Thanks Shane for bringing this up. I had no idea this problem existed. I moved it up to the top of the backlog so we can discuss prioritization in an upcoming meeting.

@chrismore let us discuss the work required to fix the bug so we can more easily prioritize. I think it should definitely be addressed first before looking into adding newsletter sign-up to sign-in.

Implementation thoughts for this feature card:
Since our sign-in success rate is a bit low and we are putting a lot of effort into increasing it, I don't feel very comfortable adding a distraction below the form, however perhaps there is an opportunity after the sign-in was successful. 

We had discussed using that opportunity to remind single device users about syncing with another device but perhaps @ryanfeeley would have a UX proposal to include newsletter opt-in too.

Thoughts?

**Comment by ryanfeeley** [comment]
@chrismore If we put the checkbox on sign-in without knowing their email first, we risk having subscribers unsubscribe. We could make it appear on sign-in pages where the username is known (force reauthentication pages, and after password is enterered incorrectly but email is known)

**Comment by chrismore** [comment]
@davismtl good point to distractions on the sign in page. Could be tested to confirm that hypothesis with a small sample a/b test. The check box could do nothing real and you just measure the impact to sign in rate and newsletter sign up rate. 

@ryanfeeley I am not following on how the checkbox would risk unsubscribing them. If the email address isn't known, you don't process the newsletter subscription until after successful authentication then only after that you send the request through basket or the salesforce connector. Is that what you mean?

FYI, in 2017, the the Firefox+You is no longer going to be a general purpose newsletter and won't be even called a newsletter. It is going to be used exclusively for onboarding and lifecycle marketing from day 1 of a user to day N with the desired outcome of retention. This is coming from the Lifecycle marketing team and with the desire to switch the onboarding from sign up to sign in, I just want to make sure that we keep the lifecycle marketing email communications healthy to justify the expense of the systems and people managing them.

**Comment by rfk** [comment]
> If we put the checkbox on sign-in without knowing their email first,
> we risk having subscribers unsubscribe

To rephrase this as a question: what should we do for users who have already signed up for the newsletter, but who leave the checkbox unchecked when they sign in?  Would we unsubscribe them?  Would they expect to be unsubscribed?
